src/admin/pages/posts/post.component.ts

```diff
@@ -35,10 +35,22 @@ export class PostComponent implements OnInit, OnDestroy {
 
     // File Upload
     private _imageToUpload = null;
+    private _imagePreviewContainer: HTMLElement;
 
     setFile({ target: { files } } = <any>{}) {
+        let imgPreviewContainer = this._imagePreviewContainer = this._document.getElementById('imagePreview');
         let [file] = files;
         this._imageToUpload = file;
+
+        let reader = new FileReader();
+
+        reader.onload = function (e: Event) {
+            let img = new Image();
+            img.src = reader.result;
+            imgPreviewContainer.appendChild(img);
+        }
+
+        reader.readAsDataURL(file);
```

Comment:
Good point. The input field restricts users from selecting non-images, but this block of code doesn't know that. I'll make the change.